### ISSUE TITLE
Example code in docs for file_picker_callback doesn't work

### ISSUE THREAD

**Issue by anabright** [tinymce/tinymce#9971]
I'm trying to implement a media file picker so that I can upload a video to a server and then have TinyMCE embed the video. Similar to how it works for images when `images_upload_url` is specified.
 
I was trying to reuse [the code provided in the interactive example in the docs](https://www.tiny.cloud/docs/tinymce/latest/image/#interactive-example) for the `file_picker_callback` option:

```
file_picker_callback: (cb, value, meta) => {
    const input = document.createElement('input');
    input.setAttribute('type', 'file');
    input.setAttribute('accept', 'image/*');

    input.addEventListener('change', (e) => {
      const file = e.target.files[0];

      const reader = new FileReader();
      reader.addEventListener('load', () => {
        /*
          Note: Now we need to register the blob in TinyMCEs image blob
          registry. In the next release this part hopefully won't be
          necessary, as we are looking to handle it internally.
        */
        const id = 'blobid' + (new Date()).getTime();
        const blobCache =  tinymce.activeEditor.editorUpload.blobCache;
        const base64 = reader.result.split(',')[1];
        const blobInfo = blobCache.create(id, file, base64);
        blobCache.add(blobInfo);

        /* call the callback and populate the Title field with the file name */
        cb(blobInfo.blobUri(), { title: file.name });
      });
      reader.readAsDataURL(file);
    });

    input.click();
  },
```

But when `input.click()` is executed, instead of opening the file browser I see this in the console:
`File chooser dialog can only be shown with a user activation.`

This example code seems to be quite old because it seems it's no longer allowed to programmatically open a file chooser via JS.

Would it be possible to update the docs with an example that works?

Or can someone provide a way (or point me in the right direction) to do the following:
1. When clicking on Insert > Media
2. Show a file picker 
3. When selecting a video, upload it to a server
4. Insert the video URL in the Insert/Edit media dialog

So, basically I'd need something similar to `images_upload_url` but for media.

Thanks,

Ana

**Comment by anabright** [comment]
It turns out I only get the `File chooser dialog can only be shown with a user activation.` message if I have Chrome DevTools open. It seems to work fine with DevTools closed. My apologies!!

For anyone interested, here's how I implemented my solution to:
1. Show file picker icon when inserting Image or Media
2. Show file browser to select an image or a video
3. Upload selected image or video to a server and then have TinyMCE embed the resulting URL

```
function uploadFile(file, tinyMCECallback) {
	const xhr = new XMLHttpRequest();
	xhr.withCredentials = false;
	xhr.open('POST', 'http://my-url/editorUpload');

	xhr.onload = () => {
		if (xhr.status < 200 || xhr.status >= 300) {
			console.error('HTTP Error: ' + xhr.status);
			return;
		}

		const json = JSON.parse(xhr.responseText);

		if (!json || typeof json.location != 'string') {
			console.error('Invalid JSON: ' + xhr.responseText);
			return;
		}
		tinyMCECallback(json.location, { title: file.name });
	};

	const formData = new FormData();
	formData.append('file', file, file.name);
	xhr.send(formData);
}

function filePickerCallback(callback, value, meta) {
	const input = document.createElement('input');
	input.setAttribute('type', 'file');
	if (meta.filetype === 'media') {
		input.setAttribute('accept', 'video/*');
	} else {
		input.setAttribute('accept', 'image/*');
	}
	input.addEventListener('change', (e) => {
		const file = e.target.files[0];
		const reader = new FileReader();
		reader.addEventListener('load', () => { uploadFile(file, callback); });
		reader.readAsDataURL(file);
	});
	input.click();
}

tinyMCE.init({
	...
	file_picker_types: 'image media',
	file_picker_callback: filePickerCallback,
	...
});
```
